File: kadane.cpp
#include<iostream>
#include<climits>
using namespace std;

int kadane(int m[], int size)
{
	int max_starting = INT_MIN, max_ending = 0;

	for (int i = 0; i < size; i++)
	{
		max_ending = max_ending + m[i];
		if (max_starting < max_ending)
			max_starting = max_ending;

		if (max_ending < 0)
			max_ending = 0;
	}
	return max_starting;
}

int main()
{
	int m[] = {-2, -5, 6, -4, -3, 8, 7, -1};
	int n = sizeof(m)/sizeof(m[0]);
	int max_sum = kadane(m, n);
	cout << "Maximum contiguous sum is: " << max_sum;
	return 0;
}

File: kadane.py
def kadane(array, arr_size):  

    maxStarting = array[0] 

    maxEnding = 0  
  
    for n in range(0, arr_size): 

        maxEnding = maxEnding + array[n]  
  
        if maxEnding < 0:  

            maxEnding = 0  
          
        elif (maxStarting < maxEnding):  

            maxStarting = maxEnding  
              
    return maxStarting  

array = [-5, -1, 2, -3, -2, 4, 6]  
 
print("Maximum Subarray Sum:", kadane(array, len(array))) 